django/website/logframe/models.py

```diff
@@ -81,6 +84,26 @@ def all_assumptions(self):
         """
         return Assumption.objects.filter(result__log_frame=self)
 
+    def get_unique_slug_name(self):
+        count = LogFrame.objects.filter(slug__startswith=self.slug[:46]).count()
+
+        max_length = LogFrame._meta.get_field('slug').max_length
+        base_slug = slugify(self.name.lower())
+        slug = base_slug[:max_length]
+
+        while LogFrame.objects.filter(slug=slug).exists():
+            # Based on https://keyerror.com/blog/automatically-generating-unique-slugs-in-django at 03/03/2017
+            count += 1
+            slug = '{0}{1:d}'.format(base_slug[:max_length - len(unicode(count))], count)
+
+        return slug
+
+    def save(self, force_insert=False, force_update=False, using=None,
+             update_fields=None):
+        if not self.pk:
```

Comment:
Using the form I was able to create a log frame with name ' ' (a single space) and then got a noreversematch error for empty slug. I'm not sure at what level we should enforce this but there should be some sanity checking on the name